manifests/config.pp

```diff
@@ -74,13 +26,15 @@
   }
 
   if $::osfamily == 'RedHat' {
-    file { '/etc/oddjobd.conf.d/oddjobd-mkhomedir.conf':
-      ensure  => 'file',
-      owner   => 'root',
-      group   => 'root',
-      mode    => '0644',
-      content => template('winbind/oddjobd-mkhomedir.conf.erb'),
-      notify  => Service[['oddjobd', 'winbind',]],
+    if ($winbind::manage_oddjob_service == true) {
```

Comment:
You can take the oddjobd notify off here and add it as a subscribe to the service like so:
```puppet
service { 'oddjobd':
  ensure    => 'running',
  enable    => true,
  subscribe => File[ '/etc/oddjobd.conf.d/oddjobd-mkhomedir.conf'],
}
```